kroxylicious-additional-filters/kroxylicious-encryption/src/main/java/io/kroxylicious/filter/encryption/AesGcmEncryptor.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.filter.encryption;
+
+import java.nio.ByteBuffer;
+import java.security.GeneralSecurityException;
+
+import javax.crypto.Cipher;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.GCMParameterSpec;
+
+class AesGcmEncryptor {
```

Comment:
We ought to note that this class is @NotThreadSafe (concurrent encryptions, concurrent decryptions, concurrent  encryption/decryption would fail) owing to the shared Cipher instance and the iv array. 

Given our use of the class will be single threaded (single instance per connection and called exclusively from the netty event loop), this is fine.